### ISSUE TITLE
Make target 'install' does nothing in cmake builds

### ISSUE THREAD

**Issue by slackjaw** [firebase/firebase-cpp-sdk#60]
### Please fill in the following fields:
Pre-built SDK from the [website](https://firebase.google.com/download/cpp) or open-source from this repo: **this repo**
Firebase C++ SDK version: **SDK 6.11.0**
Firebase plugins in use (Auth, Database, etc.): **N/A**
Additional SDKs you are using (Facebook, AdMob, etc.): **N/A** 
Platform you are using the C++ SDK on (Mac, Windows, or Linux): **Linux** 
Platform you are targeting (iOS, Android, and/or desktop):  **desktop**

### Please describe the issue here:
following a cmake build, largely according to instructions:
```
mkdir desktop_build && cd desktop_build
cmake .. -DCMAKE_INSTALL_PREFIX=/some/path
cmake --build . 
```
 running ```make install``` does nothing, even as root.  Expected behavior is to install libraries and headers at the path described by CMAKE_INSTALL_PREFIX

### Please answer the following, if applicable:
Have you been able to reproduce this issue with just the [Firebase C++ quickstarts](https://github.com/firebase/quickstart-cpp) ?

**N/A**

What's the issue repro rate? (eg 100%, 1/5 etc)
**100%**

**Comment by patm1987** [comment]
Right now the desktop SDK is in beta and not really intended for use in shipping games (see [this](https://firebase.google.com/docs/cpp/setup?platform=ios#desktop-workflow)).

As that changes, I think this would be a great change to the SDK!

**Comment by slackjaw** [comment]
Hi Patrick,
Thanks for the response - it is really good to see activity in this repo again.  Much appreciated.

If the team doesn't want to add an install target, maybe they could make available the instructions/tools they use to build releases? There must already be something that knows what headers are needed and what tree structure they need to end up in.  The problem with using the SDK as a subdirectory for cmake is that it adds a vast amount of overhead to the build process - increases the time for a single-threaded build by a factor of 200 for one project (binary release won't work due to CXX11 ABI incompatibilities)

**Comment by chkuang-g** [comment]
TL;DR; You can potentially leverage ccache to improve your build time.  Just install ccache and add those lines in CMakeList.txt as [this example](https://github.com/firebase/firebase-cpp-sdk/pull/66/commits/537c21111c78d85afecebbacc537fcd23527b6e0).

We handle linking differently for prebuilt and open-source.  For prebuilt, we have an internal build tool to include all symbols from dependent libraries to app.a.  In open-source repo, we use CMake to discover all the dependencies and link them accordingly.  If you want to link directly to the libraries built from open-source, you will need to link all dependent as well, which may not be obvious what to link.

We are currently changing our build process and potentially make some build tool available to public.  But this can take awhile.

At the meantime, you should be able to leverage ccache so that you do not need to rebuild everything everytime.  As you can see from [this commit](https://github.com/firebase/firebase-cpp-sdk/pull/66/commits/537c21111c78d85afecebbacc537fcd23527b6e0), this is how we build for our tests in the new build process.


Hope this helps!
